content/docs/user-guide/testing/best-practices/_index.md

```diff
@@ -0,0 +1,327 @@
+
+## Other Automated Test Categories in O3DE
+
+### User Interface (UI) Tests
+
+{{< caution >}}
+Automated UI tests are prone to false failures and false passes. This can lead to the anti-patterns of "when anyone changes any part of the code, they have to change all of the tests" or "this test never catches any real bugs." Due to these risks, only start investing in automated UI tests after automating lower-complexity tests.
+{{< /caution >}}
+
+While there is a significant amount of UI code in O3DE, it's recommended that automated tests avoid targeting the UI layer. UI tests are relatively complex and expensive to maintain. It is also easy to attempt writing acceptance-style tests for the UI, where robotic scripts struggle to provide useful feedback. Such tests can quickly consume more human effort to debug than to manually exercise the UI.
+
+Relying on UI tests to verify features tends to be brittle, and can prompt many confusing debugging sessions. In an ideal scenario there would already be non-UI integration tests, which fully test out the underlying logic of a feature. Then any UI tests above that layer would then not be responsible for testing the system, but only for testing how the interface reacts to simple interactions. Whenever possible, prefer implementing that lower layer of non-UI automation.
+
+### Performance/Load/Stress Tests
+
+Performance testing targets the efficiency of a process, hardware configuration, or fleet of machines. This category of tests focus on monitoring throughput and detecting breaking-points. Since the focus is on recording metrics, performance tests differ from functional tests such as unit tests. Functional tests seek an objective boolean pass/fail state. Performance tests instead log subective metrics, which require additional analysis as they change over time.
+
+O3DE's performance testing tools focus on two layers of creating metrics on the local machine:
+
+1. [Micro-benchmarks](/docs/user-guide/testing/getting-started/#googlebenchmark) of individual functions with GoogleBenchmark
+   * Similar to unit tests
+2. [Profiling markers](/docs/user-guide/testing/profiler/) annotated into different systems of the engine
+   * Similar to system tests
+
+## Which Tests are Best?
+
+Short answer: the most useful automated test is usually a unit test. Don't get paralyzed by test design decisions! Write a small test now, and revise your approach while iterating on the feature. If a bug slips through your tests, revise the tests to catch it next time.
+
+In more detail: there is often no easy answer to which types of test are the most important for a feature. Writing automated tests of each kind can help ensure the highest quality products, but all code comes with a maintenance cost. A balance of tests of each kind based on their effectiveness and resilience is recocmmended:
+
+* Unit tests should account for at least 80% of the total number of tests.
```

Comment:
80% seems arbitrary here. unit test code coverage of 80% is typically the industry standard target. 80% of number of tests is an odd target that bundles unit tests in with existence of integration and other high level tests.
would be better to say if you only have one type of test unit tests are the priority.
code coverage is a different topic to cover

Reply:
This isn't discussing coverage, only raw counts of tests.  Yes it's an arbitrary recommendation. How can it be more clear about it being arbitrary? Should this simply say "a vast majority" ?

Reply:
perhaps "the vast majority of testing should be accomplished with unit tests wherever possible". you could have supplemental links pointing to testing pyramid articles.